GetIntoTeachingApi/RateLimiting/ApiClientResolveContributor.cs

```diff
@@ -8,27 +9,25 @@ namespace GetIntoTeachingApi.RateLimiting
     public class ApiClientResolveContributor : IClientResolveContributor
     {
         private readonly IClientManager _clientManager;
-        private readonly IHttpContextAccessor _contextAccessor;
         private readonly string _clientIdHeader;
 
-        public ApiClientResolveContributor(IHttpContextAccessor contextAccessor, string clientIdHeader)
+        public ApiClientResolveContributor(string clientIdHeader)
         {
             _clientManager = new ClientManager(new Env());
-            _contextAccessor = contextAccessor;
             _clientIdHeader = clientIdHeader;
         }
 
-        public string ResolveClient()
+        public Task<string> ResolveClientAsync(HttpContext httpContext)
         {
             string clientId = null;
 
-            if (_contextAccessor.HttpContext.Request.Headers.TryGetValue(_clientIdHeader, out var value))
+            if (httpContext.Request.Headers.TryGetValue(_clientIdHeader, out var value))
             {
                 var apiKey = value.ToString().Replace("Bearer ", string.Empty);
                 clientId = _clientManager.GetClient(apiKey)?.ApiKeyPrefix;
             }
 
-            return clientId;
+            return Task.FromResult(clientId);
```

Comment:
Has this been updated to async because the update to the library expects it to be?

The `ApiClientResolveContributor` actually does I/O stuff when reading the file. Could that class be updated to async, then no need to use `Task.FromResult`?

Like the highlighted bits here:

![image](https://user-images.githubusercontent.com/47089130/123981225-14bce400-d9ba-11eb-9f8f-2771ede62d0d.png)

Reply:
Yeah the interface was updated to async; there's no benefit to making the `ClientManager.GetClient` async; its a singleton that performs the IO once on boot then we're just reading from memory in that method. In your example you would be hitting the disk on every request to load all the clients into memory and then finding the one that matches which would be more expensive.

Reply:
Ah, I see. I just saw `Task.FromResult` and wondered why